### PR TITLE
【BIT】转换能力补齐：paddle.cum[max min prod]、paddle.equal_all、paddle.[empty empty_like]

### PR DESCRIPTION
**Pull request by cangtianhuang**
### accuracy.py

感谢 @cszdrg 提出的建议：调换了生成 paddle input 和 torch input 的顺序，首先生成 paddle input，否则任务线条一的修改均无法适用

### rules.py

略微修改了 code 生成顺序，代码可读性更好（代码逻辑没变）

### paddle.cummax、paddle.cummin

查询 [torch.cummax 与 paddle.cummax 对照表](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/model_convert/convert_from_pytorch/api_difference/torch/torch.cummax.html) ，已有详细对比，发现是 `paddle 参数更多`，需要为结果设置 dtype、且 paddle 的 axis 参数允许为 None，需要特殊处理。因此决定自定义 Rule。

paddle.cummax 与 paddle.cummin 是同类 api，因此为其注册同一个 CumRule，由 paddle_api 区分。当 axis 为 None 时，展平 x，axis 设置为0。torch 的输出结果为 torch.return_types.cummax 类型的元组，为其中的 values 设置为 dtype。具体操作详见 CumRule。

### paddle.cumprod

查询 [torch.cumprod 与 paddle.cumprod 对照表](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/model_convert/convert_from_pytorch/api_difference/torch/torch.cumprod.html)，已有详细对比，发现 `两者参数数量相同`，但是 paddle 的 dim 参数允许为 None，需要特殊处理。因此决定自定义 Rule。

为 paddle.cumprod 注册 CumprodRule，当 axis 为 None 时，展平 x，axis 设置为0；设置 dtype 为 torch 的数据类型，并设定默认值（由 paddle 的默认值决定），再调用 api 直接返回即可。

### paddle.equal_all

查询 [torch.equal 与 paddle.equal_all 对照表](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/model_convert/convert_from_pytorch/api_difference/torch/torch.equal.html)，已有详细对比，发现 `两者仅参数名不同`，且 Paddle 返回 0-D bool Tensor 的后续逻辑已在 accuracy.py 中处理，不再需要我们使用 .item()。因此仅需要撰写直接映射的配置即可。

对应 torch.equal，编写 paddle_torch_args_map 进行直接映射。

### paddle.empty、paddle.empty_like

由王欢导师说明，此两者的张量值由内存随机决定，无法进行 accuracy 对比，移入 rand_apis。但也为此两个api支持了 paddle2torch 转换，测试后均能正常转换（accuracy 报错）。

## 测试结果

相关测试配置基本通过，仅有一条 paddle.cumprod 未通过 accuracy 测试，反复测试后已在 fresh_report 中报告；paddle.empty、paddle.empty_like 配置移入 paddleonly，其他配置移入 support2torch。

image

### CODE REVIEW

File: tester/paddle_to_torch/mapping.json [hunk 1]
```diff
@@ -2415,18 +2422,6 @@
...
-        "paddle_torch_args_map": {
-            "x": "input",
-            "dtype": "dtype",
-            "layout": "layout",
-            "device": "device",
-            "requires_grad": "requires_grad",
-            "memory_format": "memory_format"
-        },
-        "min_input_args": 1
-    },
```
Comment (wanghuancoder) [inline tester/paddle_to_torch/mapping.json:2429]: 从测试工具的角度来看，empty不需要对比测试。但从paddle2torch的角度来看。这里支持这个转换能力是没问题的。不需要删除。

File: tester/accuracy.py [hunk 2]
```diff
@@ -50,6 +50,45 @@ def test(self):
...
+            return
+
+        try:
+            paddle.base.core.eager._for_test_check_cuda_error()
+        except Exception as err:
+            print("[cuda error]", self.api_config.config, "\n", str(err))
+            api_config_paddle_error.write(self.api_config.config+"\n")
+            api_config_paddle_error.flush()
+            return
+
```
Comment (wanghuancoder) [inline tester/accuracy.py:91]: 这里为啥要加这段代码？

File: tester/paddle_to_torch/mapping.json (hunk 1 above)
Comment (cangtianhuang) [reply tester/paddle_to_torch/mapping.json:2429]: 我明白了，我将补齐empty两个api🫡

File: tester/accuracy.py (hunk 2 above)
Comment (cangtianhuang) [reply tester/accuracy.py:91]: @cszdrg 提出的建议：调换了生成 paddle input 和 torch input 的顺序，需要首先生成 paddle input，否则任务线条一的修改均无法适用

Comment (cangtianhuang) [reply tester/accuracy.py:91]: 即把下面那一整段 paddle 的代码挪上来了～

Comment (cszdrg) [reply tester/accuracy.py:91]: > 这里为啥要加这段代码？

因为先前的工作里，在gen_paddle_input里增加了index和key等变量操作，如果先调用gen_torch_input会导致config_analyzer里面用不了index和key等变量，产生问题

**Review by cszdrg — APPROVED** [review]

File: tester/accuracy.py [hunk 3]
```diff
@@ -50,6 +50,45 @@ def test(self):
...
+        try:
+            if not self.gen_paddle_input():
+                print("gen_paddle_input failed")
+                return
+            if "paddle.Tensor." in self.api_config.api_name:
+                api = getattr(self.paddle_args[0], self.api_config.api_name[self.api_config.api_name.rindex(".")+1:])
+                args = []
+                if len(self.paddle_args) > 1:
+                    args = self.paddle_args[1:]
+
```
Comment (wanghuancoder) [inline tester/accuracy.py:62]: 这里为什么要提前呢？因为有时候我会制造很多非法case。需要先跑Torch，证明case是有效的再跑Paddle。如果先跑Paddle，就直接Paddle报错，视作bug出报告了。

Comment (cangtianhuang) [reply tester/accuracy.py:62]: 因为任务线条一中很多 numpy tensor 的生成和 paddle 参数有关，直接 gen_torch_input 就没法用上任务一中的逻辑了。能否先仅提前 gen_paddle_input，把所需的 numpy tensor 生成出来？

**Review by wanghuancoder — APPROVED** [review]
LGTM
